### ISSUE TITLE
Too many reqs on kafka not available 

### ISSUE THREAD

**Issue by lud4ik** [aio-libs/aiokafka#496]
I have code with simple consumer. I undestand that most of time at least 1 of nodes will be available. But on kafka stop it send too many requests and uses up to 100% cpu
```
2019-04-11 12:06:55,732 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,732 Unable to update metadata from [0]
2019-04-11 12:06:55,732 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,732 Failed fetch messages from 0: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 0).
1
2019-04-11 12:06:55,733 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,733 Unable to update metadata from [0]
2019-04-11 12:06:55,734 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,734 Failed fetch messages from 0: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 0).
1
2019-04-11 12:06:55,734 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,734 Unable to update metadata from [0]
2019-04-11 12:06:55,735 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,735 Failed fetch messages from 0: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 0).
1
2019-04-11 12:06:55,736 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,736 Unable to update metadata from [0]
2019-04-11 12:06:55,736 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,736 Failed fetch messages from 0: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 0).
1
2019-04-11 12:06:55,737 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,737 Unable to update metadata from [0]
2019-04-11 12:06:55,737 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,737 Failed fetch messages from 0: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 0).
1
2019-04-11 12:06:55,738 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,738 Unable to update metadata from [0]
2019-04-11 12:06:55,738 Unable connect to node with id 0: [Errno 111] Connect call failed ('10.0.3.2', 9092)
2019-04-11 12:06:55,738 Failed fetch messages from 0: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 0).

```

Isn't retry_backoff_ms param meant for this case? Is it used on production systems?

**Comment by tvoinarovskyi** [comment]
Hmm, seems like a bug. Will try to ckeck once I get back from pto. The case definitely deserves an integration test.

**Comment by nickgashkov** [comment]
I'm facing the same issue using Docker and Docker Compose.

There're two containers, namely `python` and `kafka`. If `kafka` container dies, `python` container endlessly tries to connect. A relevant code snippet from a `python` container:

```python
import aiokafka
import asyncio


async def main():
    
    producer = aiokafka.AIOKafkaProducer(
        loop=asyncio.get_event_loop(),
        bootstrap_servers='kafka:9092',
    )
    
    await producer.start()
    
    try:
        while True:
            await asyncio.sleep(1)
            await producer.send('test', b'test-data')
    finally:
        await producer.stop()


if __name__ == '__main__':
    loop = asyncio.get_event_loop()
    loop.run_until_complete(main())
```

After killing `kafka`...
```bash
$ docker-compose stop kafka
```
...endless reconnect loop occures in a `python` container:

```
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
ERROR:aiokafka:Unable to update metadata from [1001]
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
WARNING:aiokafka.producer.sender:Got error produce response: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 1001).
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
ERROR:aiokafka:Unable to update metadata from [1001]
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
WARNING:aiokafka.producer.sender:Got error produce response: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 1001).
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
ERROR:aiokafka:Unable to update metadata from [1001]
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
WARNING:aiokafka.producer.sender:Got error produce response: NodeNotReadyError: Attempt to send a request to node which is not ready (node id 1001).
ERROR:aiokafka:Unable connect to node with id 1001: [Errno 111] Connect call failed ('172.28.0.3', 9092)
```

**Comment by tvoinarovskyi** [comment]
Will try to find time for this on 0.6.0, as it's a bad one. We can't have logs filling like crazy.

**Comment by arvindjangra1996** [comment]
hey, can we catch "NodeNotReadyError" exception while consumer from Kafka. So that we can take any action at this situation. Like sending alert.

**Comment by tvoinarovskyi** [comment]
@arvindjangra1996 There is currently no such hook for connection errors. If you need it, please open a feature request issue, thanks!

**Comment by tvoinarovskyi** [comment]
@arvindjangra1996 Will answer in https://github.com/aio-libs/aiokafka/issues/612

**Comment by tvoinarovskyi** [comment]
Merge and released as part of 0.6.0. Hope it solves the issue.

**Comment by Olegt0rr** [comment]
@tvoinarovskyi, looks like this issue become actual again :( 

Please, help to review PR, that should help to avoid reconnect cpu flooding
https://github.com/aio-libs/aiokafka/pull/1014

Look at CPU throttling on kafka become unavailable:
<img width="919" alt="Image" src="https://github.com/user-attachments/assets/9af5935b-39bf-4b49-83b0-a414cb967b8e" />
